### DISCUSSION TITLE
Namespaced Scoped Managed Resources in Crossplane

### DISCUSSION THREAD

**Discussion by arielsepton** [crossplane/crossplane#4178] [Q&A] [answered]
Hey 👋

I have a question regarding the current absence of namespaced scoped managed resources in Crossplane. Could you please shed some light on the reasons behind this design decision?

Thank you for your time 🙂

**Comment by phisco** [comment, marked as answer]
Maybe this could be helpful: https://docs.crossplane.io/latest/concepts/composition/#claiming-composite-resources.
My understanding is that MRs are not namespaced because Claims are, similarly to what happens for PVs and PVCs.

**Reply by arielsepton** [reply]
Thanks! It was very helpful :)

**Reply by excalq** [reply]
To me this feels like an added layer (with quite a lot of complexity considering patching and compositions) to work around an underlying design limitation. Managed Resources are extremely easy to provision with Crossplane, however their cluster-only scoping precludes an obvious means to organize and group them. 

For better maintainability and safety, even if using Compositions and Claims, it'd still be preferable to have the option of namespace-scoped Managed Resources.

**Reply by patrykorwat** [reply]
In cases of environments with a strong namespace separation enforcement, this is an issue. It does take a significant effort to adopt Crossplane with this design in place as the design breaks main premise of namespaces to create _virtual clusters_ (ref https://kubernetes.io/blog/2016/08/kubernetes-namespaces-use-cases-insights/)

**Comment by JohnLBevan** [comment]
In this video it's mentioned that Crossplane v2 will introduce namespace scoped resources: https://www.youtube.com/watch?v=8zQtpcxmdhs (at 27:10)
